storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -708,8 +708,17 @@ bool xbcloud_put(Object_store *store, const std::string &container,
   }
 
   // check if xtrabackup_info, almost the last file exists in backup
-  if (std::count(object_list.begin(), object_list.end(),
-                 backup_name + "/xtrabackup_info.00000000000000000000") == 0) {
+  std::string last_file_expecting = backup_name + "/xtrabackup_info";
+  bool file_found = false;
+  for (auto cur_file : object_list) {
```

Comment:
Please add a comment here that we have two possibilities for the last filename  xtrabackup_info.gz.00000000000 or xtrabackup_info.00000000000000000000 and we try to match the prefix "backup/xtrabackup_info" to determine if xbcloud is successful or not